adapter/pkg/logging/package_logs.go

```diff
@@ -26,6 +26,23 @@ import (
 	"github.com/wso2/product-microgateway/adapter/pkg/config"
 )
 
+// ErrorLog used to keep error details for error logs
+type ErrorLog struct {
+	Message  string
+	Severity string
+	Code     int
+}
+
+// Log represents the extended type of logrus.logger
+type Log struct {
+	*logrus.Logger
+}
+
+// ErrorC can be used for formal error logs
+func (l *Log) ErrorC(e ErrorLog) {
+	l.WithFields(logrus.Fields{"severity": e.Severity, "code": e.Code}).Errorf(e.Message)
```

Comment:
Constants should be added for field names "severity" & "error_code"

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/2633/commits/c3087100180167139f73a7fb351a91359c5d9bbf